_studio/mfx_lib/shared/src/mfx_common_int.cpp

```diff
@@ -116,37 +112,41 @@ mfxStatus CheckFrameInfoCommon(mfxFrameInfo  *info, mfxU32 /* codecId */)
         case MFX_FOURCC_Y216:
         case MFX_FOURCC_Y416:
 #endif
-
             break;
 
         default:
-            return MFX_ERR_INVALID_VIDEO_PARAM;
+            MFX_RETURN(MFX_ERR_INVALID_VIDEO_PARAM);
         }
     }
 
-    if (info->Shift)
+    switch (info->FourCC)
     {
-        if (   info->FourCC != MFX_FOURCC_P010 && info->FourCC != MFX_FOURCC_P210
+    case MFX_FOURCC_P010:
+    case MFX_FOURCC_P210:
 #if (MFX_VERSION >= 1027)
-            && info->FourCC != MFX_FOURCC_Y210
+    case MFX_FOURCC_Y210:
 #endif
 #if (MFX_VERSION >= 1031)
-            && info->FourCC != MFX_FOURCC_P016 && info->FourCC != MFX_FOURCC_Y216
-            && info->FourCC != MFX_FOURCC_Y416
+    case MFX_FOURCC_P016:
+    case MFX_FOURCC_Y216:
+    case MFX_FOURCC_Y416:
 #endif
-            )
-            return MFX_ERR_INVALID_VIDEO_PARAM;
+        // HEVC HW support both shifts, check rest of cases
+        if (!(isHW && codecId == MFX_CODEC_HEVC))
```

Comment:
As I understand, the prev. logic was - 'the `Shift` can be set only for the certain FOURCCs'. It doesn't restrict to have `Shift` both values (zero/one) and I see no reasons to put codec's specific logic (HEVC) right here. I'd rather put it either to component's code or to `CheckFrameInfoCodecs` at least, where we already have both `codecId` and `isHW`.

Reply:
@dgurulev , it seems `CheckFrameInfoCodecs()` already has all useful checks. So I'll abandon separate 
[HEVCd] supports both kind of shifts